v2/src/log_context_property_type_wchar_t_ptr.c

```diff
@@ -0,0 +1,148 @@
+    (void)value;
+}
+
+static LOG_CONTEXT_PROPERTY_TYPE wchar_t_ptr_log_context_property_type_get_type(void)
+{
+    /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_012: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).get_type shall return the property type LOG_CONTEXT_PROPERTY_TYPE_wchar_t_ptr. ]*/
+    return LOG_CONTEXT_PROPERTY_TYPE_wchar_t_ptr;
+}
+
+int LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr)(void* dst_value, size_t count, const wchar_t* format, ...)
+{
+    int result;
+
+    if (
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_013: [ If dst_value is NULL, LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr) shall fail and return a non-zero value. ]*/
+        (dst_value == NULL) ||
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_014: [ If format is NULL, LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr) shall fail and return a non-zero value. ]*/
+        (format == NULL)
+        )
+    {
+        (void)printf("Invalid arguments: void* dst_value=%p, const wchar_t* format=%ls\r\n",
+            dst_value, MU_WP_OR_NULL(format));
+        result = MU_FAILURE;
+    }
+    else
+    {
+        va_list args;
+        va_start(args, format);
+
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_015: [ LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr) shall initialize by calling vswprintf the memory at dst_value with the wprintf style formatted wchar_t string given by format and the arguments in .... ]*/
+        int vswprintf_result = vswprintf(dst_value, count, format, args);
+
+        if (vswprintf_result < 0)
+        {
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_017: [ If formatting the string fails, LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr) shall fail and return a non-zero value. ]*/
+            result = MU_FAILURE;
+        }
+        else
+        {
+            /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_016: [ LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr) shall succeed and return 0. ]*/
+            result = 0;
+        }
+
+        va_end(args);
+    }
+
+    return result;
+}
+
+int LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(wchar_t_ptr)(const wchar_t* format, ...)
+{
+    va_list args;
+    va_start(args, format);
+
+    wchar_t buffer[LOG_MAX_WCHAR_STRING_LENGTH];
+    /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_018: [ If vswprintf fails, LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(wchar_t_ptr) shall return a negative value. ]*/
+    /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_019: [ Otherwise, on success, LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(wchar_t_ptr) shall return the amount of memory needed in number of wide-chracters to store the wprintf style formatted wchar_t string given by format and the arguments in .... ]*/
+    int result = vswprintf(buffer, LOG_MAX_WCHAR_STRING_LENGTH, format, args) * sizeof(wchar_t);
```

Comment:
> vswprintf(buffer, LOG_MAX_WCHAR_STRING_LENGTH, [](http://example.com/codeflow?start=16&length=47)

I think the intention here is to only get the size. We might then call vswprintf with NULL, 0 for the first and second parameter. I suspect that the work done by `vswprintf` would be less then.

Reply:
pass NULL and 0 doesn't work on linux, leave it as before